domain/schema/model/sql/0019-application.sql

```diff
@@ -200,6 +200,55 @@ CREATE TABLE application_status (
     REFERENCES workload_status_value (id)
 );
 
+CREATE TABLE device_constraint (
+    uuid TEXT NOT NULL PRIMARY KEY,
+    application_uuid TEXT NOT NULL,
+    name TEXT NOT NULL,
+    type_id INT,
+    count INT,
+    attribute_uuid TEXT,
+    CONSTRAINT fk_device_constraint_application
+    FOREIGN KEY (application_uuid)
+    REFERENCES application (uuid),
+    CONSTRAINT fk_device_constraint_device_type
+    FOREIGN KEY (type_id)
+    REFERENCES device_constraint_type (id),
+    CONSTRAINT fk_device_constraint_attribute
+    FOREIGN KEY (attribute_uuid)
+    REFERENCES device_constraint_attribute (uuid)
+);
+
+CREATE TABLE device_constraint_type (
+    id INT PRIMARY KEY,
+    value TEXT
+);
+
+INSERT INTO device_constraint_type VALUES
+(0, 'gpu'),
+(1, 'nvidia.com/gpu'),
+(2, 'amd.com/gpu');
+
```

Comment:
These were not enforced in code but I think they should be in our DDL, see https://github.com/juju/juju/blob/main/core/devices/constraints.go#L21-L26.

Reply:
I think we should leave it as text and not constrain it.

Reply:
looks like these are defined in https://kubernetes.io/docs/tasks/manage-gpus/scheduling-gpus/#using-device-plugins

Reply:
This is not a good idea for a couple of reasons:

 1. Having a vendor name (AMD and Nvidia)  in juju database is a smell without thought. What happens if Nvidia goes away in the next 10 years (the support lifetime). I'm just picking a contrived example.
 2. We should define a better representation. This has bitten us before; LXD Profiles is a clear example of inferior modeling.

We can sidestep this for now. Passing it as raw text, we can start defining better semantics around it in the future. This wasn't thought enough about when adding it to the codebase.

Reply:
Yeah I agree, we can always add this back later if needed. Removed it and turned it into `TEXT NOT NULL`.